pkg/provider/configuration.go

```diff
@@ -223,6 +280,26 @@ func BuildTCPRouterConfiguration(ctx context.Context, configuration *dynamic.TCP
 	}
 }
 
+// BuildUDPRouterConfiguration Builds a router configuration.
+func BuildUDPRouterConfiguration(ctx context.Context, configuration *dynamic.UDPConfiguration) {
+	for routerName, router := range configuration.Routers {
+		loggerRouter := log.FromContext(ctx).WithField(log.RouterName, routerName)
+
+		if len(router.Service) == 0 {
+			if len(configuration.Services) > 1 {
+				delete(configuration.Routers, routerName)
+				loggerRouter.
+					Error("Could not define the service name for the router: too many services")
+				continue
+			}
+
+			for serviceName := range configuration.Services {
+				router.Service = serviceName
+			}
```

Comment:
I know it is not needed at the moment, because we already check the number of services above, but just for clarity, and to make sure no regression ever happens later:

```suggestion
				break
			}
```